### ISSUE TITLE
SSHCluster multiple worker options

### ISSUE THREAD

**Issue by BENR0** [dask/distributed#4714]
Currently when using `SSHCluster` worker options are the same on all nodes. To a certain degree this limits the usage on clusters with heterogeneous nodes (e.g. some nodes have 4 cpu's while other might have more). So it would be nice if similar to the connect options different worker options could be specified for each node.

Additionally it would be nice to optionally be able to specify custom names for each node instead of the automatic generated number names.

The proposed changes should not change the current behaviour so should be backwards compatible and are mainly related to the lines:

https://github.com/dask/distributed/blob/74a9aff7e7429c4b2f12204cfa97aba1fca5aa29/distributed/deploy/ssh.py#L357-L366

Proposed change:
````Python
"options": {
                "address": host,
                "connect_options": connect_options
                if isinstance(connect_options, dict)
                else connect_options[i + 1],
                "kwargs": worker_options
                if isinstance(worker_options, dict)
                else worker_options[i],
                "worker_module": worker_module,
                "remote_python": remote_python[i + 1]
                if isinstance(remote_python, list)
                else remote_python,
                "name": names[i]
                if isinstance(names, list)
                else None,
            },
````

`names` would have to be added as a new argument.

**Comment by jacobtomlinson** [comment]
Supporting heterogenous workers is definitely interesting. This would definitely require some thinking about as we may want to add labels and things to enable task annotation.

For the CPU problem specifically you can now set `nprocs` as `auto` which should detect local resources the same way `LocalCluster` does.

**Comment by BENR0** [comment]
@jacobtomlinson `nprocs` is definitely interesting even though I can't find it in the documentation (https://docs.dask.org/en/latest/setup/single-distributed.html#localcluster). Unfortunately I have a very custom use case and also want to set the number of threads. I think that is not possible is it?

@jrbourbeau Thanks for the info I haven't seen the `new_worker_name` method I just oriented the proposed change on how it is done for the connect options.

In any case I think case it would be nice to have more flexibility when working with heterogenous workers. Obviously I am not very familiar with the code but if I can help in any way to bring this on the way let me know.

**Comment by jacobtomlinson** [comment]
@BENR0 you're right that this is not as configurable as you need.

A while ago we changed `connect_options` to also take a list of options, one per connection, to allow different usernames and things for SSH connections. https://github.com/dask/distributed/blob/94254184bb4d81eca914baf374ce0763488d327e/distributed/deploy/ssh.py#L271-L276

Perhaps we should do the same for `worker_options`?

**Comment by BENR0** [comment]
@jacobtomlinson yes that's exactly what I was proposing. Sorry if that was not clear.

**Comment by jacobtomlinson** [comment]
Ah excellent, then yes that would be great. Do you think you could raise a PR for this?

**Comment by BENR0** [comment]
Yes I already have something link this "hacked" together for personal use. I will sort it out and make a PR when I find time. Since I proposed it I guess it is not a pressing matter :-D.

**Comment by johanvandegriff** [comment]
Would this allow for using a different SSH port for each host, or should I open a separate issue for that?

**Comment by BENR0** [comment]
@johanvandegriff no this is only for dask related settings of the workers. For ssh related things you can use "connect_options". See https://docs.dask.org/en/latest/setup/ssh.html for the documentation.

**Comment by jacobtomlinson** [comment]
@johanvandegriff this is already possible today with `connect_options` as @BENR0 linked.

**Comment by gcasella** [comment]
I'm very interested in this as well, some of my code that uses dask requires me to define specific workers. 

For example when using dask-worker cli I would define a name like "worker00" with the options of --nprocs 8 --nthreads 1 and I would need just one of these to do my work such as "worker00-0"
